frontend_tests/node_tests/emoji.js

```diff
@@ -78,15 +78,43 @@ zrequire('util');
 }());
 
 (function test_translate_emoticons_to_names() {
-  global.emoji_codes = {
-      codepoint_to_name: {
-          '1f603': 'smiley',
-      },
-  };
+    // Simple test
+    var test_text = 'Testing :)';
+    var expected = 'Testing :smiley:';
+    var result = emoji.translate_emoticons_to_names(test_text);
+    assert.equal(expected, result);
 
-  var test_text = 'Testing :)';
-  var expected = 'Testing :smiley:';
-  var result = emoji.translate_emoticons_to_names(test_text);
-
-  assert.equal(expected, result);
+    // Extensive tests.
+    // The following code loops over the test cases and each emoticon conversion
+    // to generate multiple test cases.
+    var EMOTICON_CONVERSIONS = {
+        ':)': 'smiley',
+        '(:': 'smiley',
+        ':(': 'slightly_frowning_face',
+        '<3': 'heart',
+        ':|': 'expressionless',
+        ':/': 'confused',
+    };
+    var testcases = [
+        {name: 'only emoticon', original: ':original:', expected: ':converted:'},
```

Comment:
I think this would read clearer if you used syntax like `<original>` rather than using `:`s as the delimiters; because we have emoji which use `:`s as the delimeter involved here